storage/src/lib.rs

```diff
@@ -22,120 +22,192 @@ const ALEO_DIRECTORY: &str = ".aleo";
-        true => match std::env::current_dir() {
-            Ok(current_dir) => current_dir,
-            _ => PathBuf::from(env!("CARGO_MANIFEST_DIR")),
-        },
-        // In production mode, the ledger is stored in the `~/.aleo/` directory.
-        false => aleo_dir(),
-    };
+    let mut path = aleo_base_storage_path(dev);
 
     // Construct the path to the ledger in storage.
     match dev {
         // In development mode, the ledger files are stored in a hidden folder.
         Some(id) => {
-            path.push(format!(".ledger-{}-{}", network, id));
-            path
+            path.push(format!("ledger-{}-{}", network, id));
         }
         // In production mode, the ledger files are stored in a visible folder.
         None => {
-            path.push("storage");
             path.push(format!("ledger-{}", network));
-            path
         }
     }
+
+    path
 }
 
 ///
-/// Returns the directory for accessing the operator files from Aleo storage.
+/// Returns the directory for accessing the prover files from Aleo storage.
 ///
-/// In production mode, the expected directory path is `~/.aleo/storage/operator-{network}`.
-/// In development mode, the expected directory path is `/path/to/repo/.operator-{network}-{id}`.
+/// The expected paths are:
 ///
-pub fn aleo_operator_dir(network: u16, dev: Option<u16>) -> PathBuf {
-    // Retrieve the starting directory.
-    let mut path = match dev.is_some() {
-        // In development mode, the operator is stored in the repository root directory.
-        true => match std::env::current_dir() {
-            Ok(current_dir) => current_dir,
-            _ => PathBuf::from(env!("CARGO_MANIFEST_DIR")),
-        },
-        // In production mode, the operator is stored in the `~/.aleo/` directory.
-        false => aleo_dir(),
-    };
+/// - Production: `~/.aleo/storage/prover-{network}`.
+/// - Development: `../.aleo/storage/prover-{network}-{id}`.
+///
+/// Note: in development mode, the `.aleo` directory will be created in the cargo workspace as
+/// opposed to the home directory, see [`aleo_dir`] for details.
+///
+pub fn aleo_prover_dir(network: u16, dev: Option<u16>) -> PathBuf {
+    let mut path = aleo_base_storage_path(dev);
+
+    // Construct the path to the prover in storage.
+    match dev {
+        // In development mode, the prover files are stored in a hidden folder.
+        Some(id) => {
+            path.push(format!("prover-{}-{}", network, id));
+        }
+        // In production mode, the prover files are stored in a visible folder.
+        None => {
+            path.push(format!("prover-{}", network));
+        }
+    }
+
+    path
+}
 
-    // Construct the path to the operator in storage.
+///
+/// Returns the path for the primary-related BFT files.
+///
+/// The expected paths are:
+///
+/// - Production: `~/.aleo/storage/bft-{network}/primary`.
+/// - Development: `../.aleo/storage/bft-{network}/primary-{id}`.
```

Comment:
```suggestion
/// - Development: `<snarkOS>/.aleo/storage/bft-{network}/primary-{dev_id}`.
```

Reply:
also, I guess we should follow the ledger and prover logic and do `<snarkOS>/.aleo/storage/bft-{network}-{dev_id}/primary` instead, no?

Reply:
Good catch!